packages/legend-application-query/src/components/Core_LegendQueryApplicationPlugin.tsx

```diff
@@ -463,24 +465,85 @@ export class Core_LegendQueryApplicationPlugin extends LegendQueryApplicationPlu
                     }
                   }
                 }
+              } else if (editorStore instanceof MappingQueryCreatorStore) {
```

Comment:
Do we really need to navigate users ? 
I would expect us to just clear the current query if they are not in  `ExistingQueryEditorStore`.
We should also add a blocking modal if there are changes in their query.